trackers_integration/issuetracker/openproject.py

```diff
@@ -164,17 +183,34 @@ def _report_issue(self, execution, user):
             project_id, getattr(settings, "OPENPROJECT_WORKPACKAGE_TYPE_NAME", "Bug")
         )
 
-        new_issue = self.rpc.create_workpackage(
-            project_id,
-            {
-                "subject": f"Failed test: {execution.case.summary}",
-                "description": {"raw": self._report_comment(execution, user)},
-                "_links": {
-                    "type": _type["_links"]["self"],
-                },
-            },
+        category = self.get_workpackage_category(
+            project_id, execution.case.category.name
         )
 
+        if category:
+            new_issue = self.rpc.create_workpackage(
+                project_id,
+                {
+                    "subject": f"Failed test: {execution.case.summary}",
+                    "description": {"raw": self._report_comment(execution, user)},
+                    "_links": {
+                        "type": _type["_links"]["self"],
+                        "category": category["_links"]["self"],
```

Comment:
I don't particularly like the duplicated RPC call inside the two branches of this if condition. 

What happens if you pass `"category": None` ?

Reply:
Alternatively you can store the second argument in a dict variable and add the `category` key only if needed and then pass the variable to the RPC call.